src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -1034,18 +1034,17 @@ private JsValue CmpXchangeInternal(string key)
             {
                 if (string.IsNullOrEmpty(key))
                     return JsValue.Undefined;
-                BlittableJsonReaderObject value = null;
+
                 using (_database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
                 using (ctx.OpenReadTransaction())
                 {
-                    value = _database.ServerStore.Cluster.GetCompareExchangeValue(ctx, key).Value;
-                }
-
-                if (value == null)
-                    return null;
+                    var value = _database.ServerStore.Cluster.GetCompareExchangeValue(ctx, key).Value;
+                    if (value == null)
+                        return null;
 
-                var jsValue = TranslateToJs(ScriptEngine, _jsonCtx, value);
-                return jsValue.AsObject().Get("Object");
+                    var jsValue = TranslateToJs(ScriptEngine, _jsonCtx, value.Clone(_jsonCtx));
+                    return jsValue.AsObject().Get("Object");
+                }
```

Comment:
Why do we have to extend this to 'TranslateToJs'. The `documentsContext.CloseTransaction();` is not enough in `ExecuteQuery` ?

Reply:
Those are two different transaction: one for the compEx access and the other for the document access. We accessed here a blittable who'se context was disposed (the .CloseTransaction and the TranlateToJS changes are two different fixes to two different problems, that end up "meeting" in that scenario)

Reply:
Right. Thanks!